### PR TITLE
Support kubernetes service deployer

### PR DESCRIPTION
**Pull request by mtojek**
## What does this PR do?

This PR defines files used by the Kubernetes service deployer.

## Why is it important?

These files will define a custom service deployed on top of the Kubernetes cluster. The Elastic Agent can monitor this service (e.g. nginx deployed in the cluster). If there aren't any files required to be installed in the cluster, it's mandatory to set the `.empty` file as a mark for the test runner to use Kubernetes service deployer (empty directory `_dev/deploy/k8s` won't be versioned).

## Checklist

- [x] I have added test packages to [`code/go/internal/validator/test/packages`](https://github.com/elastic/package-spec/tree/master/code/go/internal/validator/test/packages) that prove my change is effective.
- [x] I have added an entry in [`versions/N/changelog.yml`](https://github.com/elastic/package-spec/blob/master/versions/1/changelog.yml).

## Related issues

- https://github.com/elastic/elastic-package/issues/239

### CODE REVIEW

**Comment by mtojek** [comment]
@ycombinator is ready for review, but please take a look at the https://github.com/elastic/elastic-package/pull/246 to verify if the proposal here is correct.

File: code/go/internal/validator/folder_item_content.go [hunk 1]
```diff
@@ -17,8 +17,8 @@ func loadItemContent(itemPath, mediaType string) ([]byte, error) {
 		return nil, errors.Wrap(err, "reading item file failed")
 	}
 
-	if len(itemData) == 0 {
-		return nil, errors.New("file is empty")
+	if len(itemData) == 0 && mediaType != "" {
```
Comment (ycombinator) [inline code/go/internal/validator/folder_item_content.go:20]: It took me a second to realize that this additional check was required to properly handle the `.empty` file case. In the case of the `.empty` file, `len(itemData) == 0` but `mediaType == ""` so this check will evaluate to `false`, which is what we want. Without the additional `mediaType != ""` clause, this check would have evaluated to `true` and returned an error, which is not what we want.

Would you mind adding a comment above this check explaining why we need to check both the length of `itemData` and the media type?

**Review by ycombinator — APPROVED** [review]
Left a minor comment about adding an explanatory comment. 🙂 

Otherwise, LGTM.

File: code/go/internal/validator/folder_item_content.go (hunk 1 above)
Comment (mtojek) [reply code/go/internal/validator/folder_item_content.go:23]: Fixed.

Comment (mtojek) [reply code/go/internal/validator/folder_item_content.go:23]: I'll merge it once the CI is green and doesn't complain.

**Pull request merged**